_aicspylibczi/ImageFactory.h

```diff
@@ -11,19 +12,34 @@ namespace pylibczi {
       using PixelType = libCZI::PixelType;
       using CtorMap = std::map<libCZI::PixelType,
                                std::function<std::shared_ptr<Image>(
-                                   std::vector<size_t>, libCZI::PixelType pixel_type_, const libCZI::CDimCoordinate* plane_coordinate_,
-                                   libCZI::IntRect box_, int mIndex_)
+                                   std::vector<size_t>, libCZI::PixelType pixel_type_,
+                                   const libCZI::CDimCoordinate* plane_coordinate_,
+                                   libCZI::IntRect box_, ImagesContainerBase* bptr,
+                                   size_t img_index_, int mIndex_)
                                > >;
       using SplitCtorMap = std::map<libCZI::PixelType, std::function<std::shared_ptr<Image>( std::shared_ptr<Image> img_, int channel_) > >;
 
       static CtorMap s_pixelToImage;
-      static SplitCtorMap s_pixelToSplit;
+
+      ImagesContainerBase::ImagesContainerBasePtr m_imgContainer;
```

Comment:
It looks like this class was formerly 100% static.  Why does it need this member variable now?  Could the imgContainer be passed in from the outside to keep all the member functions static?

Reply:
My thinking was to let the ImageFactory handle all the logic of memory, essentially making it a controller and to keep any memory logic (outside of the pybind11 object conversion) contained in the ImageFactory. Is there an advantage to having all the functions being static? I assumed that essentially just mean the class is a namespace?

Reply:
Yes, it has huge implications for whether you have to instantiate the class or can just call its functions without creating objects.  But it all depends on your desired usage pattern.

Reply:
I get what you are saying. In this case, I don't think it has much impact. The alternative is to add the memory object to every function call and the cost is the class being stateful but it is only being used in a stateful context anyway.